sdk/eventhub/event-hubs/CHANGELOG.md

```diff
@@ -2,6 +2,9 @@
 
 ## 5.2.1 (Unreleased)
 
+- Fixes issue [#8584](https://github.com/Azure/azure-sdk-for-js/issues/8584)
+  where a `TypeError` was sometimes thrown as an uncaught exception.
```

Comment:
We are doing more than just fix #8584 here right? The refresh connection is a fix ported over from Service Bus. Can we provide some more details on what changes we are making and how the user can benefit from it?

Reply:
I added the note we had in service-bus.
Also updated core-amqp in service-bus and event-hubs to the latest version.

Reply:
```suggestion
  where attempting to create AMQP links when the AMQP connection was in the
  process of closing resulted in a `TypeError` in an uncaught exception.
```